packages/react-native-web/src/exports/Image/index.js

```diff
@@ -123,27 +126,50 @@ function resolveAssetUri(source): ?string {
           : prev
       );
     }
+
     const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
-    uri = asset
-      ? `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`
-      : '';
+    const uri = `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`;
+    resolvedSource = { uri, width: asset.width, height: asset.height };
   } else if (typeof source === 'string') {
-    uri = source;
+    resolvedSource.uri = source;
+  } else if (Array.isArray(source)) {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'The <Image> component does not support multiple sources passed as array, falling back to the first source in the list',
+        { source }
+      );
+    }
+
+    return resolveSource(source[0]);
   } else if (source && typeof source.uri === 'string') {
-    uri = source.uri;
+    const { uri, width, height, headers } = source;
+    resolvedSource = { uri, width, height, headers };
```

Comment:
This kind of goes with the previous comment, but what properties are you trying to strip and why can't they remain?

Reply:
The `source` object at this point resolves to the following shape 

https://github.com/Expensify/react-native-web/blob/bd1f7b887ad690ef926f0c3d9f97c1f625d4bd55/packages/react-native-web/src/exports/Image/types.js#L23-L78

This is a non-exact type, which means it can carry additional fields 

We want to make sure we resolve a `resolvedSource` complying to the `ImageSource` exact type

https://github.com/Expensify/react-native-web/blob/6a5d237a40db9f03fe4118e3d805b07ae10edc46/packages/react-native-web/src/modules/ImageLoader/index.js#L12-L17

And so we destruct only the necessary fields

The reason we want to comply with a `ImageSource` type is to make the rest of the code simpler

E.g. `source` can change but this should not trigger image loading unless any of the image loading fields changed
We achieve this by returning the same resolved source, because any arbitrary fields would be stripped